Lib/test/test_math.py

```diff
@@ -2503,6 +2503,43 @@ def test_input_exceptions(self):
         self.assertRaises(TypeError, math.atan2, 1.0)
         self.assertRaises(TypeError, math.atan2, 1.0, 2.0, 3.0)
 
+    def test_exception_messages(self):
+        x = -1.1
+        with self.assertRaises(ValueError,
+                               msg=f"expected a nonnegative input, got {x}"):
+            math.sqrt(x)
+        with self.assertRaises(ValueError,
+                               msg=f"expected a positive input, got {x}"):
+            math.log(x)
+        with self.assertRaises(ValueError,
+                               msg=f"expected a positive input, got {x}"):
```

Comment:
Would it be feasible to make the error message say which input is nonpositive, since there are two parameters?

Reply:
It's possible, but I'm not sure if it worth code complications.  After all, I doubt that 2-arg form of log() is popular enough.

On another hand, what if instead of
```pycon
>>> math.log(2,-2)
Traceback (most recent call last):
  File "<python-input-1>", line 1, in <module>
    math.log(2,-2)
    ~~~~~~~~^^^^^^
ValueError: expected a positive input
```
we have
```
>>> math.log(2,-2)
Traceback (most recent call last):
  File "<python-input-1>", line 1, in <module>
    math.log(2,-2)
    ~~~~~~~~   ^^
ValueError: expected a positive input
```
?

Input validation is a common pattern and in such cases we want to point on some specific argument of the function.

Reply:
That would be nice, but it would probably require some machinery that we don't currently have.

Reply:
FYI: https://github.com/python/cpython/issues/124819